pkg/clustermesh/common/remote_cluster.go

```diff
@@ -189,6 +192,11 @@ func (rc *remoteCluster) restartRemoteConnection() {
 					return err
 				}
 
+				if err := rc.validateConnection(config); err != nil {
+					rc.getLogger().WithError(err).Error("Terminating remote cluster connection due to invalid remote cluster configuration")
+					rc.releaseOldConnection()
+					return err
+				}
```

Comment:
I'd personally include this validation along that of the ClusterID for consistency (https://github.com/cilium/cilium/blob/eb1f266a5da54d19300e38a7e9a48c2d53852b89/pkg/clustermesh/remote_cluster.go#L68-L72), and perform it only in the clustermesh case (as kvstoremesh doesn't really need it).